components/common/src/ui.rs

```diff
@@ -130,18 +132,18 @@ pub trait UIReader {
     fn prompt_yes_no(&mut self, question: &str, default: Option<bool>) -> Result<bool>;
 }
 
+pub trait ColorPrinter {
```

Comment:
Why define our own trait when we could use [`termcolor::WriteColor`](https://docs.rs/termcolor/1.0.4/termcolor/trait.WriteColor.html)?

Since our streams will implement `Write` as well, we can use the standard [`write`](http://doc.rust-lang.org/1.32.0/std/macro.write.html) and [`writeln`](http://doc.rust-lang.org/1.32.0/std/macro.writeln.html) macros.

Reply:
Having a one liner color printing call seemed much simpler. I ended up adding this trait for the sake of [ctrl-gateway/src/mod.rs](https://github.com/habitat-sh/habitat/pull/6203/files#diff-25e976e0941144b82133d34113753ac4). Not sure how you would handle its needs and implement `WriteColor`.

Reply:
I'll defer to your judgement for what gives better ergonomics in the end. The nice thing about building upon `termcolor::WriteColor` is that we could keep a single code path for all our printing and just enable/disable colors with [`ColorChoice`](https://docs.rs/termcolor/1.0.4/termcolor/enum.ColorChoice.html).

If we find code like this
```rust
buffer.set_color(ColorSpec::new().set_fg(Some(Color::Green)))?;
writeln!(&mut buffer, "green text!")?;
bufwtr.print(&buffer)?;
```
to be too verbose, there's no reason we can't build helpers on top of it like:
```rust
our_helper.print("green text!", Color::Green, Weight::Normal)?;
```
or
```rust
our_helper.set_color(Green).print("green text!")?;
```
while still retaining the logic-unifying goodness of `termcolor::WriteColor` in the underlying implementation.

Reply:
Yeah my first draft used `WriteColor` but that breaks down in the `ctrl-gateway` when sending colorized messages through the `ConsoleLine` protocol where the actual coloring and underlying console API calls need to happen on the gateway client.

Reply:
Can you show me a little more specifically where your approach ran into trouble?

Reply:
I was hoping that the `termcolor::Buffer` would be serializable but its not :( Maybe you know of a way to send a `Buffer` through prost but I think it has to implement `Serializable`

Reply:
I'm thinking of this more and I think I see how to do this with WriteColor. I'll play with it.

Reply:
Sounds great. Let me know if you want input on anything.